dxr/build.py

```diff
@@ -523,6 +522,11 @@ def docs():
                     total['annotations'] = annotations_for_this_line
                 yield es.index_op(total)
 
+                # Because needles_by_line holds a reference, total is not
```

Comment:
So total points to needles_by_line but nothing points to total. So how is total not getting GC'd?

Ah, I see: total is a pointer into a member of needles_by_line, which we mutably inflate by updating it with the file-wide needles. It makes me a little uncomfortable to go around destructively clear()ing pieces of a closed-over variable in a function that might not obviously be expected to. So I think we should either make a loud note about that in docs()'s docstring or else take a different tack: make a shallow copy of the member of needles_by_line and put that into `total` as our first step in the loop. (If the latter, I'd be curious how much it hurts speed.)